crates/symbolicator/src/services/symbolication.rs

```diff
@@ -1845,14 +1843,24 @@ fn find_stackwalking_problem(
     // Normalize the name of the `eflags` register (returned by breakpad) to `efl` (returned by rust-minidump).
     // Not doing this leads to tons of spurious diffs.
     let mut stacktraces_breakpad = result_breakpad.stacktraces.clone();
+    let mut scan = false;
     for stacktrace in stacktraces_breakpad.iter_mut() {
         if let Some(val) = stacktrace.registers.remove("eflags") {
             stacktrace.registers.insert("efl".to_string(), val);
         }
+
+        if !scan {
+            for frame in stacktrace.frames.iter() {
+                if frame.trust == FrameTrust::Scan {
+                    scan = true;
+                    break;
+                }
+            }
+        }
```

Comment:
this is all rather convoluted, no?  Isn't this the same without a `mut scan`?

```
let scan = stracktraces_breakpad.iter()
               .flat_map(|st| st.frames.iter())
               .any(|f| f.trust == FrameTrust::Scan);
```

Reply:
I did it this way so as to only have one loop over `stacktraces_breakpad`. But that's probably a pointless micro-optimization and doing it this way is clearer.

Reply:
you could also compromise and only use `.any()` on the inner iteration leaving the `mut scan` around.  but as you say, it's a micro-optimisation and clearer code probably matters more.